File: examples/python/lib/fhir_loader.py
"""Load FHIR bundles from JSON files."""

from dataclasses import dataclass
from pathlib import Path

from fhir.resources.R4B.bundle import Bundle

from .constants import DATA_SOURCES

DATA_DIR = Path(__file__).parent.parent.parent.parent / "data" / "fhir-exports"


@dataclass
class DataSource:
    name: str
    bundle: Bundle


def load_all_bundles() -> list[DataSource]:
    """Load all FHIR bundles from the data directory."""
    return [
        DataSource(
            name=source["name"],
            bundle=Bundle.model_validate_json((DATA_DIR / source["file"]).read_text()),
        )
        for source in DATA_SOURCES
    ]

File: examples/nextjs/src/lib/fhir-loader.ts
/// <reference types="@types/fhir" />

import { promises as fs } from "fs";
import path from "path";

export interface DataSource {
  name: string;
  bundle: fhir4.Bundle;
}

const DATA_DIR = path.join(process.cwd(), "..", "..", "data", "fhir-exports");

const DATA_SOURCES = [
  { name: "Metro General Hospital", file: "metro-general-hospital.json" },
  { name: "CityCare Primary Clinic", file: "citycare-clinic.json" },
  { name: "HealthFirst Laboratories", file: "healthfirst-labs.json" },
];

export async function loadAllBundles(): Promise<DataSource[]> {
  return Promise.all(
    DATA_SOURCES.map(async (source) => ({
      name: source.name,
      bundle: JSON.parse(
        await fs.readFile(path.join(DATA_DIR, source.file), "utf-8")
      ) as fhir4.Bundle,
    }))
  );
}
